python/cudaq/kernel/analysis.py

```diff
@@ -52,38 +52,45 @@ def getVariableName(self, node):
             return self.getVariableName(node.value)
         return ''
 
+    def checkForMeasureResult(self, value):
```

Comment:
Is there a reason we can't use the `quake-add-metadata` pass to determine if it has mid-circuit measurements? It seems like some (or all?) of this is duplicated logic from `lib/Optimizer/Transforms/QuakeAddMetadata.cpp`. It's possible that Python needs to know this earlier than C++, and if so, then perhaps running that pass would be too late, but I figured I would ask. My very quick scan of the Python code's use of `hasMidCircuitMeasures`/ `conditionalOnMeasure` makes it seem like its whole job in life is setting `qubitMeasurementFeedback`, and that aforementioned pass does the same thing.

Reply:
I have the same question. We should not duplicate this functionality in Python unless there is an extremely compelling argument why the pass can't be run.

Reply:
Great question! I know this is invoked before the compile to MLIR stage in Python, but I will confirm if it can be moved down to make use of the existing pass.

Reply:
I think I will have to compile all the kernels beforehand to make use of the `quake-add-metadata` function pass in Python workflow instead of the `MidCircuitMeasurementAnalyzer` class.
@amccaskey: Can you confirm this?

Reply:
This analysis pass may actually be a relic from the original implementation - originally we had a library mode for Python and this pass was necessary to inform the runtime that the kernel had mid circuit measurement. We can probably drop it now for the MLIR pass. But you will have to update `sample.py` to check the Module attribute instead of the python kernel metadata attribute.

Reply:
WIP...

Reply:
> WIP...

IMO, if it's unreasonable to get this done before the release, I'd be ok deferring this work until after the release. I'd hate to rush some refactoring at the last minute, as that could add its own risk to the release.

Reply:
Sounds good, I will make the refactoring a separate PR.